components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1686,6 +1690,13 @@ private void handleRequestObject(OAuthMessage oAuthMessage, OAuthAuthzRequest oa
              */
         overrideAuthzParameters(oAuthMessage, parameters, oauthRequest.getParam(REQUEST),
                 oauthRequest.getParam(REQUEST_URI), requestObject);
+
+        // If the redirect uri was not given in auth request the registered redirect uri will be available here,
+        // so validating if the registered redirect uri is a single uri that can be properly redirected.
+        if (StringUtils.startsWith(parameters.getRedirectURI(), REGEX_PATTERN)) {
```

Comment:
```suggestion
        if (StringUtils.isBlank(parameters.getRedirectURI()) || StringUtils.startsWith(parameters.getRedirectURI(), REGEX_PATTERN)) {
```
lets add this as, if the redirect_uri is null/blank then there should be a error